libsolidity/analysis/TypeChecker.cpp

```diff
@@ -2656,6 +2656,9 @@ bool TypeChecker::visit(MemberAccess const& _memberAccess)
 				_memberAccess.location(),
 				"Using \"." + memberName + "(...)\" is deprecated. Use \"{" + memberName + ": ...}\" instead."
 			);
+
+		if (!funType->bound())
+			annotation.requiresVirtualLookup = !!dynamic_cast<ContractType const*>(exprType);
```

Comment:
Can you name examples for both cases?

Reply:
As far as I understand:
```
contract A {
    function f() virtual public {}
}
contract B is A {
    function f() virtual override public {}
    function x() public { 
        A.f(); // This would be a `false` case because the type would be `TypeType` 
        this.f(); // This would be a `true` case because `this` should be the `ContractType` 
}
```

Reply:
Actually `this.f()` is an external call and we should not do virtual lookup for external calls - so both of them should be non-virtual.

Reply:
I guess valid lookups in ``this`` will have non-empty ``possibleMembers`` and not end up here at all? But yeah, it's a bit non-transparent so far :-)...

Reply:
I think that's also my problem, I don't know all those cases. I basically derived this behavior based on the backend `resolveVirtual` and `superFunction` calls